Consul/Health.cs

```diff
@@ -284,11 +298,22 @@ public Task<QueryResult<ServiceEntry[]>> Service(string service, string tag, boo
         /// <param name="passingOnly">Only return if the health check is in the Passing state</param>
         /// <param name="q">Customized query options</param>
         /// <param name="filter">Specifies the expression used to filter the queries results prior to returning the data</param>
+        /// <param name="healthType">Specifies the type of health check to query for</param>
         /// <param name="ct">Cancellation token for long poll request. If set, OperationCanceledException will be thrown if the request is cancelled before completing</param>
         /// <returns>A query result containing the service members matching the provided service ID, tag, and health status, or a query result with a null response if no service members matched the filters provided</returns>
-        public Task<QueryResult<ServiceEntry[]>> Service(string service, string tag, bool passingOnly, QueryOptions q, Filter filter, CancellationToken ct = default)
+        public Task<QueryResult<ServiceEntry[]>> Service(string service, string tag, bool passingOnly, QueryOptions q, Filter filter, string healthType, CancellationToken ct = default)
         {
-            var req = _client.Get<ServiceEntry[]>(string.Format("/v1/health/service/{0}", service), q, filter);
+            string path;
+            switch (healthType)
+            {
+                case "connectHealth":
+                    path = "/v1/health/connect/{0}";
+                    break;
+                default:
+                    path = "/v1/health/service/{0}";
+                    break;
+            }
+            var req = _client.Get<ServiceEntry[]>(string.Format(path, service), q, filter);
```

Comment:
I know that the documentation mentions that the `connect` and `service` endpoints are nearly identical, but I don't think it's a good idea to rely on that. I would rather split this into two separate implementations, one for `connect` and one for `service` endpoints.

Reply:
Okei noted, so instead of implementing the switch case, I should implement as a separate entity?

Reply:
> Okei noted, so instead of implementing the switch case, I should implement as a separate entity?

yes, I think it is going to be better that way.